keras/optimizers.py

```diff
@@ -12,7 +12,28 @@
 
 def clip_norm(g, c, n):
     if c > 0:
-        g = K.switch(n >= c, g * c / n, g)
+        if K.backend() == 'tensorflow':
```

Comment:
Add a comment explaining why we need to add shape inference code in the TF case.